gabrielmulleer/teste-oak-tecnologia
# Feature: Registration Form

## Summary

This pull request introduces a product registration page for entering new product details.

## Changes

- Created `(main)/page.tsx` containing the product registration form.
- Implemented `(main)/_components/form.tsx` for the form component.
- Added form fields: product name, description, price, and availability.
- Implemented basic form handling and state management.
- Added `api/products/route.ts` to define API endpoints and `POST` functionality.
- Introduced `(main)/schema.ts` for form schema and validation using the `zod` library.

## Testing

1. Run `npm run dev` to launch the development server.
2. Navigate to the appropriate route.
3. Verify the presence of the product registration form with all specified fields.
4. Complete the form and submit it to observe the logged output in the console (simulating product saving logic).

## Notes

This pull request focuses on establishing the API structure and form functionality. Future updates will include refinements to the user interface and additional features.
